tests/test_client_request.js

```diff
@@ -108,11 +108,15 @@ describe('Direct use of `ClientRequest`', () => {
 
   it('should emit an expected error when no interceptors and net connect is disallowed', done => {
     nock.disableNetConnect()
-    new http.ClientRequest({ port: 12345, path: '/' }).on('error', err => {
-      expect(err.message).to.equal(
-        'Nock: Disallowed net connect for "localhost:12345/"'
-      )
-      done()
-    })
+    const request = new http.ClientRequest({ port: 12345, path: '/' }).on(
+      'error',
+      err => {
+        expect(err.message).to.equal(
+          'Nock: Disallowed net connect for "localhost:12345/"'
+        )
+        done()
+      }
+    )
+    request.end()
```

Comment:
That is technically a breaking change because the error is not thrown immediately, but I think this is the correct behavior and the way this test should have been written in the first place